### ISSUE TITLE
converting tf.keras model to onnx  using tf2onnx adding extra layers

### ISSUE THREAD

**Issue by Ashish0091** [onnx/onnx#5850]
I successfully trained a CNN model for text recognition, and it performs well during inference. However, upon converting the model from Keras to ONNX, I observed alterations in the architecture after a few layers. Notably, I noticed the inclusion of additional layers that were not part of the original Keras model.

followings are the keras and onnx model visualization in netron after maxpool layer (marked in red)
![keras_netron](https://github.com/onnx/onnx/assets/24604443/671321a4-bcd9-4c19-8f6c-aeb59e6b8661)
![onnx_netron](https://github.com/onnx/onnx/assets/24604443/2b08bcb8-0c21-429c-8763-2baab46f3e07)

model conversion script: 
[https://github.com/Ashish0091/fcn_text_recognition/blob/master/fcn2onnx_export.py](url)

keras model details:
`
class CTCLayer(keras.layers.Layer):
    def __init__(self, name=None):
        super().__init__(name=name)
        self.loss_fn = keras.backend.ctc_batch_cost
    def call(self, y_true, y_pred):
        batch_len = tf.cast(tf.shape(y_true)[0], dtype="int64")
        input_length = tf.cast(tf.shape(y_pred)[1], dtype="int64")
        label_length = tf.cast(tf.shape(y_true)[1], dtype="int64")
        input_length = input_length * tf.ones(shape=(batch_len, 1), dtype="int64")
        label_length = label_length * tf.ones(shape=(batch_len, 1), dtype="int64")
        loss = self.loss_fn(y_true, y_pred, input_length, label_length)
        self.add_loss(loss)
        return y_pred

def build_model():
    input_img = keras.Input(shape=(image_width, image_height, 1), name="image")
    labels = keras.layers.Input(name="label", shape=(None,))
    x = keras.layers.Conv2D(
        32,
        (3, 3),
        activation="relu",
        kernel_initializer="he_normal",
        padding="same",
        name="Conv1",
    )(input_img)
    x = keras.layers.MaxPooling2D((2, 2), name="pool1")(x)
    x = keras.layers.Conv2D(
        64,
        (3, 3),
        activation="relu",
        kernel_initializer="he_normal",
        padding="same",
        name="Conv2",
    )(x)
    x = keras.layers.MaxPooling2D((2, 2), name="pool2")(x)
    new_shape = ((image_width // 4), (image_height // 4) * 64)
    x = keras.layers.Reshape(target_shape=new_shape, name="reshape")(x)
    x = keras.layers.Dense(64, activation="relu", name="dense1")(x)
    x = keras.layers.Dropout(0.2)(x)
    x = keras.layers.Dense(len(char_to_num.get_vocabulary()) + 2, activation="softmax", name="dense2")(x)
    output = CTCLayer(name="ctc_loss")(labels, x)
    model = keras.models.Model(
    inputs=[input_img, labels], outputs=output, name="handwriting_recognizer")
    opt = keras.optimizers.Adam()
    model.compile(optimizer=opt)
    return model
`

Model name: CNN for text recognition 
onnx opset=11
keras version = 2.14.0
tensorflow version=2.14.0
onnx version = 1.15.0
onnxruntime = 1.16.3

Note: I have used different opset version to check whether these extra layers are added or not.

**Comment by xadupre** [comment]
It seems an issue for https://github.com/onnx/tensorflow-onnx.

**Comment by justinchuby** [comment]
They are probably semantically equivalent?
